Security definitions and principles expansion (#677)
bitwarden/contributing-docs: docs/architecture/security/principles/01-servers-are-zero-knowledge.mdx

is fully isolated from the server and infrastructure; it cannot be read nor expanded outside of the
user's client context.

This is what we mean when we sometimes refer to "end-to-end encrypted" or "zero knowledge".

## Threat model

- An attacker able to view and/or edit database data
- An attacker able to create, edit, and fully control network responses to a client

Some examples of attackers included in this list could be Bitwarden employees, network
administrators, CDNs, and any MITM that can circumvent TLS protection.

## Security goals

- Attackers cannot retrieve decrypted vault data
- Attackers cannot retrieve user encryption keys
- Attackers cannot inject arbitrary items as vault data
- Attackers cannot edit vault data\*

> _\*We do not guarantee availability of vault data in this threat model as attackers have full

Reviewer: ```suggestion
_\* Availability of vault data is not a security goal as attackers have full
```
